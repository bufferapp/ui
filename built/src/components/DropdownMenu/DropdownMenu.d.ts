import React from 'react';
export default class DropdownMenu extends React.Component {
    constructor(props: any);
    componentDidMount(): void;
    componentWillUnmount(): void;
    handleMousedown: () => void;
    togglePopup: () => void;
    isPopupOpen: () => any;
    openPopup: () => void;
    closePopup: () => void;
    handleKeydown: (event: any) => void;
    handlePopupBlur: (event: any) => void;
    render(): React.JSX.Element;
}
//# sourceMappingURL=DropdownMenu.d.ts.map