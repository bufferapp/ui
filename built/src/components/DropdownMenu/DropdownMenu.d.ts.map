{"version":3,"file":"DropdownMenu.d.ts","sourceRoot":"","sources":["../../../../src/components/DropdownMenu/DropdownMenu.tsx"],"names":[],"mappings":"AACA,OAAO,KAAK,MAAM,OAAO,CAAA;AAMzB,MAAM,CAAC,OAAO,OAAO,YAAa,SAAQ,KAAK,CAAC,SAAS;gBAE3C,KAAK,KAAA;IAgBjB,iBAAiB;IAOjB,oBAAoB;IAOpB,eAAe,aAEd;IAED,WAAW,aAIV;IAGD,WAAW,YAA0B;IAErC,SAAS,aAER;IAED,UAAU,aAET;IAGD,aAAa,uBA8BZ;IAGD,eAAe,uBAOd;IAED,MAAM;CAgEP"}