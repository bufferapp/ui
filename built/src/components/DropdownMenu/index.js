export { default } from './DropdownMenu';
//# sourceMappingURL=index.js.map