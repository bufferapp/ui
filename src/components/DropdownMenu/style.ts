import styled, { css } from 'styled-components'
import {
  blue,
  white,
  gray,
  grayDefault,
  boxShadow,
  grayLight,
  grayDarker,
} from '../style/colors'
import {
  fontFamily,
  fontSize,
  lineHeight,
  fontWeightMedium,
} from '../style/fonts'
import { borderRadius } from '../style/borders'
import { ORG_SWITCHER } from '../NavBar/NavBar'

export const DropdownItems = styled.ul`
  padding: 0;
  margin: 0;
  display: flex;
  height: 100%;

  &[role='menubar'] > li {
    display: flex;
    position: relative;

    & [role='menuitem']:focus {
      outline: ${(props) => (props.usingMouse ? 'none' : `1px solid ${blue}`)};
      outline-offset: -1px;
    }
  }
`

export const Item = styled.li`
  font-family: ${fontFamily};
  font-weight: ${fontWeightMedium};
  font-size: 14px;
  line-height: 16px;
  position: ${(props) => (props.type === ORG_SWITCHER ? 'relative' : '')};
  min-height: 32px;
  align-items: center;
  color: ${(props) => (props.disabled ? grayLight : gray)};
  cursor: pointer;
  display: flex;
  text-overflow: ellipsis;
  user-select: none;
  white-space: nowrap;
  background-color: transparent;
  padding: ${(props) => (props.menuOption ? 0 : '0 8px')};
  margin: ${(props) => {
    const { type } = props
    return type === 'header' ? '0 0 5px 5px' : ''
  }};
`

export const ItemDividerTitle = styled.span`
  position: relative;
  top: -11px;
  margin-left: 8px;
  background: white;
  padding: 0 8px;
  font-size: 12px;
  font-family: ${fontFamily};
  color: ${gray};
`

const buttonBase = css`
  -webkit-appearance: none;
  align-items: center;
  background-color: ${white};
  border: 0;
  border-radius: ${borderRadius};
  color: ${grayDefault};
  cursor: pointer;
  display: flex;
  justify-content: flex-start;
  flex: 0 0 auto;
  font-family: ${fontFamily};
  font-size: ${fontSize};
  font-weight: ${fontWeightMedium};
  line-height: ${lineHeight};
  min-width: 0;
  padding-top: 0;
  padding-bottom: 0;
  position: relative;
  transition-property: background-color, border-color, color;
  transition-duration: 0.1s;
  transition-timing-function: ease-in;
  user-select: none;
  :active {
    top: 1px;
  }
  :focus {
    box-shadow: 0 0 0 3px ${boxShadow};
  }
  :hover {
    transition-property: background-color, border-color, color;
    transition-duration: 0.1s;
    transition-timing-function: ease-in;
  }
`

const medium = css`
  padding-right: 16px;
  padding-left: 16px;
  height: 40px;
`

const disabled = css`
  background-color: ${grayLight};
  cursor: not-allowed;
`

// @ts-expect-error TS(7006) FIXME: Parameter 'props' implicitly has an 'any' type.
const getColor = (props, type, defaultColor) => {
  if (props.colors && props.colors[type]) {
    return props.colors[type]
  }
  return defaultColor
}

export const ButtonItemStyled = styled.button`
  ${buttonBase};
  ${medium};
  color: ${(props) => getColor(props, 'title', grayDarker)};
  background: none transparent;
  border-radius: 0;
  justify-content: flex-start;
  border-radius: 4px;
  height: 32px;
  width: 100%;
  padding: 0px 8px;
  margin: 0;
  :focus {
    box-shadow: none;
  }
  :hover {
    background-color: ${grayLight};
    svg {
      fill: ${(props) => getColor(props, 'iconHover', grayDarker)};
    }
  }
  ${(props) => (props.disabled ? disabled : '')};
`

export const ButtonLabel = styled.div`
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  margin-left: ${(props) => {
    const { hasIcon, type } = props
    if (type === ORG_SWITCHER) {
      return hasIcon ? '5px' : '21px'
    }
    return hasIcon ? '5px' : '0px'
  }};
  margin-right: ${(props) => (props.hasIcon ? '5px' : '0px')};
`
