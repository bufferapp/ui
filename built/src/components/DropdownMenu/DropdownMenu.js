import React from 'react';
import PropTypes from 'prop-types';
import { DropdownItems, Item } from './style';
import PopupMenu from './PopupMenu/PopupMenu';
import { keyCode } from './keyCode';
export default class DropdownMenu extends React.Component {
    constructor(props) {
        super(props);
        this.handleMousedown = () => {
            this.setState({ usingMouse: true });
        };
        this.togglePopup = () => {
            const { isOpen } = this.state;
            this.setState({ isOpen: !isOpen });
        };
        this.isPopupOpen = () => this.state.isOpen;
        this.openPopup = () => {
            this.setState({ isOpen: true });
        };
        this.closePopup = () => {
            this.setState({ isOpen: false });
        };
        this.handleKeydown = (event) => {
            let flag = false;
            switch (event.keyCode) {
                case this.keyCode.SPACE:
                case this.keyCode.RETURN:
                case this.keyCode.DOWN:
                    if (!this.isPopupOpen()) {
                        this.openPopup();
                        flag = true;
                    }
                    break;
                case this.keyCode.ESC:
                case this.keyCode.TAB:
                    this.closePopup();
                    break;
                default:
                    break;
            }
            this.setState({ usingMouse: false });
            if (flag) {
                event.stopPropagation();
                event.preventDefault();
            }
        };
        this.handlePopupBlur = (event) => {
            const outsideOfPopup = !event.currentTarget.contains(event.relatedTarget);
            setTimeout(() => {
                if (this.isPopupOpen() && outsideOfPopup) {
                    this.closePopup();
                }
            }, 300);
        };
        this.state = {
            isOpen: false,
            usingMouse: false,
        };
        this.keyCode = keyCode;
        this.handleKeydown = this.handleKeydown.bind(this);
        this.handlePopupBlur = this.handlePopupBlur.bind(this);
        this.handleMousedown = this.handleMousedown.bind(this);
    }
    componentDidMount() {
        this.itemsNode.addEventListener('keydown', this.handleKeydown);
        this.itemsNode.addEventListener('mousedown', this.handleMousedown);
    }
    componentWillUnmount() {
        this.itemsNode.removeEventListener('keydown', this.handleKeydown);
        this.itemsNode.removeEventListener('mousedown', this.handleMousedown);
    }
    render() {
        const { menubarItem, items, ariaLabel, ariaLabelPopup, horizontalOffset, xPosition, onOpen, } = this.props;
        const MenubarItem = React.cloneElement(menubarItem);
        return (React.createElement(DropdownItems, { ref: (itemsNode) => (this.itemsNode = itemsNode), role: "menubar", "aria-label": ariaLabel, usingMouse: this.state.usingMouse },
            React.createElement(Item, { role: "none", menuOption: true },
                React.createElement(MenubarItem.type, Object.assign({}, MenubarItem.props, { role: "menuitem", tabIndex: "0", "aria-haspopup": "true", "aria-expanded": this.state.isOpen, onKeyDown: (ev) => this.handleKeydown(ev), onClick: (ev) => {
                        this.togglePopup();
                        ev.preventDefault();
                    } })),
                React.createElement(PopupMenu, { role: "menu", xPosition: xPosition, items: items, "aria-label": ariaLabelPopup, horizontalOffset: horizontalOffset, isOpen: this.state.isOpen, usingMouse: this.state.usingMouse, closePopup: this.closePopup, onBlur: (event) => this.handlePopupBlur(event), onOpen: onOpen }))));
    }
}
DropdownMenu.propTypes = {
    menubarItem: PropTypes.node.isRequired,
    ariaLabel: PropTypes.string.isRequired,
    ariaLabelPopup: PropTypes.string,
    xPosition: PropTypes.string,
    items: PropTypes.arrayOf(PropTypes.shape({
        id: PropTypes.string,
        title: PropTypes.string,
        colors: PropTypes.shape({
            title: PropTypes.string,
            iconHover: PropTypes.string,
        }),
    })).isRequired,
    onOpen: PropTypes.func,
};
DropdownMenu.defaultProps = {
    ariaLabelPopup: null,
    xPosition: 'right',
    onOpen: null,
};
//# sourceMappingURL=DropdownMenu.js.map