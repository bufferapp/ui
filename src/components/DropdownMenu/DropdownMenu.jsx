/* eslint-disable jsx-a11y/role-supports-aria-props */
import React from 'react';
import PropTypes from 'prop-types';
import { DropdownItems, Item } from './style';
import PopupMenu from './PopupMenu/PopupMenu';
import { keyCode } from './keyCode';

export default class DropdownMenu extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      isOpen: false,
      usingMouse: false,
    };

    this.keyCode = keyCode;

    this.handleKeydown = this.handleKeydown.bind(this);
    this.handlePopupBlur = this.handlePopupBlur.bind(this);
    this.handleMousedown = this.handleMousedown.bind(this);
  }

  componentDidMount() {
    this.itemsNode.addEventListener('keydown', this.handleKeydown);
    this.itemsNode.addEventListener('mousedown', this.handleMousedown);
  }

  componentWillUnmount() {
    this.itemsNode.removeEventListener('keydown', this.handleKeydown);
    this.itemsNode.removeEventListener('mousedown', this.handleMousedown);
  }

  handleMousedown = () => {
    this.setState({ usingMouse: true });
  };

  togglePopup = () => {
    const { isOpen } = this.state;
    this.setState({ isOpen: !isOpen });
  };

  isPopupOpen = () => this.state.isOpen;

  openPopup = () => {
    this.setState({ isOpen: true });
  };

  closePopup = () => {
    this.setState({ isOpen: false });
  };

  handleKeydown = (event) => {
    let flag = false;
    switch (event.keyCode) {
      case this.keyCode.SPACE:
      case this.keyCode.RETURN:
      case this.keyCode.DOWN:
        if (!this.isPopupOpen()) {
          this.openPopup();
          flag = true;
        }
        break;
      case this.keyCode.ESC:
      case this.keyCode.TAB:
        this.closePopup();
        break;
      default:
        break;
    }

    this.setState({ usingMouse: false });

    if (flag) {
      event.stopPropagation();
      event.preventDefault();
    }
  };

  handlePopupBlur = (event) => {
    const outsideOfPopup = !event.currentTarget.contains(event.relatedTarget);
    setTimeout(() => {
      if (this.isPopupOpen() && outsideOfPopup) {
        this.closePopup();
      }
    }, 300);
  };

  render() {
    const {
      menubarItem,
      items,
      ariaLabel,
      ariaLabelPopup,
      horizontalOffset,
      xPosition,
      onOpen,
    } = this.props;

    const MenubarItem = React.cloneElement(menubarItem);

    return (
      <DropdownItems
        ref={(itemsNode) => (this.itemsNode = itemsNode)}
        role="menubar"
        aria-label={ariaLabel}
        usingMouse={this.state.usingMouse}
      >
        <Item role="none" menuOption>
          <MenubarItem.type
            {...MenubarItem.props}
            role="menuitem"
            tabIndex="0"
            aria-haspopup="true"
            aria-expanded={this.state.isOpen}
            onKeyDown={(ev) => this.handleKeydown(ev)}
            onClick={(ev) => {
              this.togglePopup();
              ev.preventDefault();
            }}
          />
          <PopupMenu
            role="menu"
            xPosition={xPosition}
            items={items}
            aria-label={ariaLabelPopup}
            horizontalOffset={horizontalOffset}
            isOpen={this.state.isOpen}
            usingMouse={this.state.usingMouse}
            closePopup={this.closePopup}
            onBlur={(event) => this.handlePopupBlur(event)}
            onOpen={onOpen}
          />
        </Item>
      </DropdownItems>
    );
  }
}

DropdownMenu.propTypes = {
  /** Link in the NavBar that triggers the popup */
  menubarItem: PropTypes.node.isRequired,

  /** Aria label for list component */
  ariaLabel: PropTypes.string.isRequired,

  /** Aria label for popup menu, it should preferibly be the same as the menubarItem name */
  ariaLabelPopup: PropTypes.string,

  /** xPosition for Dropdown menu */
  xPosition: PropTypes.string,

  /** Items list to display in the popup menu */
  items: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string,
      title: PropTypes.string,
      colors: PropTypes.shape({
        title: PropTypes.string,
        iconHover: PropTypes.string,
      }),
    })
  ).isRequired,

  /** onOpen function to fire when the Dropdown menu is open */
  onOpen: PropTypes.func,
};

DropdownMenu.defaultProps = {
  ariaLabelPopup: null,
  xPosition: 'right',
  onOpen: null,
};
