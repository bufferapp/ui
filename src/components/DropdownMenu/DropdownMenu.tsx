/* eslint-disable jsx-a11y/role-supports-aria-props */
import React from 'react'
import PropTypes from 'prop-types'
import { DropdownItems, Item } from './style'
import PopupMenu from './PopupMenu/PopupMenu'
import { keyCode } from './keyCode'

export default class DropdownMenu extends React.Component {
  // @ts-expect-error TS(7006) FIXME: Parameter 'props' implicitly has an 'any' type.
  constructor(props) {
    super(props)

    this.state = {
      isOpen: false,
      usingMouse: false,
    }

    // @ts-expect-error TS(2339) FIXME: Property 'keyCode' does not exist on type 'Dropdow... Remove this comment to see the full error message
    this.keyCode = keyCode

    this.handleKeydown = this.handleKeydown.bind(this)
    this.handlePopupBlur = this.handlePopupBlur.bind(this)
    this.handleMousedown = this.handleMousedown.bind(this)
  }

  componentDidMount() {
    // @ts-expect-error TS(2339) FIXME: Property 'itemsNode' does not exist on type 'Dropd... Remove this comment to see the full error message
    this.itemsNode.addEventListener('keydown', this.handleKeydown)
    // @ts-expect-error TS(2339) FIXME: Property 'itemsNode' does not exist on type 'Dropd... Remove this comment to see the full error message
    this.itemsNode.addEventListener('mousedown', this.handleMousedown)
  }

  componentWillUnmount() {
    // @ts-expect-error TS(2339) FIXME: Property 'itemsNode' does not exist on type 'Dropd... Remove this comment to see the full error message
    this.itemsNode.removeEventListener('keydown', this.handleKeydown)
    // @ts-expect-error TS(2339) FIXME: Property 'itemsNode' does not exist on type 'Dropd... Remove this comment to see the full error message
    this.itemsNode.removeEventListener('mousedown', this.handleMousedown)
  }

  handleMousedown = () => {
    this.setState({ usingMouse: true })
  }

  togglePopup = () => {
    // @ts-expect-error TS(2339) FIXME: Property 'isOpen' does not exist on type 'Readonly... Remove this comment to see the full error message
    const { isOpen } = this.state
    this.setState({ isOpen: !isOpen })
  }

  // @ts-expect-error TS(2339) FIXME: Property 'isOpen' does not exist on type 'Readonly... Remove this comment to see the full error message
  isPopupOpen = () => this.state.isOpen

  openPopup = () => {
    this.setState({ isOpen: true })
  }

  closePopup = () => {
    this.setState({ isOpen: false })
  }

  // @ts-expect-error TS(7006) FIXME: Parameter 'event' implicitly has an 'any' type.
  handleKeydown = (event) => {
    let flag = false
    switch (event.keyCode) {
      // @ts-expect-error TS(2339) FIXME: Property 'keyCode' does not exist on type 'Dropdow... Remove this comment to see the full error message
      case this.keyCode.SPACE:
      // @ts-expect-error TS(2339) FIXME: Property 'keyCode' does not exist on type 'Dropdow... Remove this comment to see the full error message
      case this.keyCode.RETURN:
      // @ts-expect-error TS(2339) FIXME: Property 'keyCode' does not exist on type 'Dropdow... Remove this comment to see the full error message
      case this.keyCode.DOWN:
        if (!this.isPopupOpen()) {
          this.openPopup()
          flag = true
        }
        break
      // @ts-expect-error TS(2339) FIXME: Property 'keyCode' does not exist on type 'Dropdow... Remove this comment to see the full error message
      case this.keyCode.ESC:
      // @ts-expect-error TS(2339) FIXME: Property 'keyCode' does not exist on type 'Dropdow... Remove this comment to see the full error message
      case this.keyCode.TAB:
        this.closePopup()
        break
      default:
        break
    }

    this.setState({ usingMouse: false })

    if (flag) {
      event.stopPropagation()
      event.preventDefault()
    }
  }

  // @ts-expect-error TS(7006) FIXME: Parameter 'event' implicitly has an 'any' type.
  handlePopupBlur = (event) => {
    const outsideOfPopup = !event.currentTarget.contains(event.relatedTarget)
    setTimeout(() => {
      if (this.isPopupOpen() && outsideOfPopup) {
        this.closePopup()
      }
    }, 300)
  }

  render() {
    const {
      // @ts-expect-error TS(2339) FIXME: Property 'menubarItem' does not exist on type 'Rea... Remove this comment to see the full error message
      menubarItem,
      // @ts-expect-error TS(2339) FIXME: Property 'items' does not exist on type 'Readonly<... Remove this comment to see the full error message
      items,
      // @ts-expect-error TS(2339) FIXME: Property 'ariaLabel' does not exist on type 'Reado... Remove this comment to see the full error message
      ariaLabel,
      // @ts-expect-error TS(2339) FIXME: Property 'ariaLabelPopup' does not exist on type '... Remove this comment to see the full error message
      ariaLabelPopup,
      // @ts-expect-error TS(2339) FIXME: Property 'horizontalOffset' does not exist on type... Remove this comment to see the full error message
      horizontalOffset,
      // @ts-expect-error TS(2339) FIXME: Property 'xPosition' does not exist on type 'Reado... Remove this comment to see the full error message
      xPosition,
      // @ts-expect-error TS(2339) FIXME: Property 'onOpen' does not exist on type 'Readonly... Remove this comment to see the full error message
      onOpen,
    } = this.props

    const MenubarItem = React.cloneElement(menubarItem)

    return (
      <DropdownItems
        // @ts-expect-error TS(2339) FIXME: Property 'itemsNode' does not exist on type 'Dropd... Remove this comment to see the full error message
        ref={(itemsNode) => (this.itemsNode = itemsNode)}
        role="menubar"
        aria-label={ariaLabel}
        // @ts-expect-error TS(2769) FIXME: No overload matches this call.
        usingMouse={this.state.usingMouse}
      >
        {/* @ts-expect-error TS(2769) FIXME: No overload matches this call. */}
        <Item role="none" menuOption>
          <MenubarItem.type
            {...MenubarItem.props}
            role="menuitem"
            // @ts-expect-error TS(2322) FIXME: Type 'string' is not assignable to type 'number'.
            tabIndex="0"
            aria-haspopup="true"
            // @ts-expect-error TS(2339) FIXME: Property 'isOpen' does not exist on type 'Readonly... Remove this comment to see the full error message
            aria-expanded={this.state.isOpen}
            onKeyDown={(ev) => this.handleKeydown(ev)}
            onClick={(ev) => {
              this.togglePopup()
              ev.preventDefault()
            }}
          />
          <PopupMenu
            // @ts-expect-error TS(2322) FIXME: Type '{ role: string; xPosition: any; items: any; ... Remove this comment to see the full error message
            role="menu"
            xPosition={xPosition}
            items={items}
            aria-label={ariaLabelPopup}
            horizontalOffset={horizontalOffset}
            // @ts-expect-error TS(2339) FIXME: Property 'isOpen' does not exist on type 'Readonly... Remove this comment to see the full error message
            isOpen={this.state.isOpen}
            // @ts-expect-error TS(2339) FIXME: Property 'usingMouse' does not exist on type 'Read... Remove this comment to see the full error message
            usingMouse={this.state.usingMouse}
            closePopup={this.closePopup}
            // @ts-expect-error TS(7006) FIXME: Parameter 'event' implicitly has an 'any' type.
            onBlur={(event) => this.handlePopupBlur(event)}
            onOpen={onOpen}
          />
        </Item>
      </DropdownItems>
    )
  }
}

// @ts-expect-error TS(2339) FIXME: Property 'propTypes' does not exist on type 'typeo... Remove this comment to see the full error message
DropdownMenu.propTypes = {
  /** Link in the NavBar that triggers the popup */
  menubarItem: PropTypes.node.isRequired,

  /** Aria label for list component */
  ariaLabel: PropTypes.string.isRequired,

  /** Aria label for popup menu, it should preferibly be the same as the menubarItem name */
  ariaLabelPopup: PropTypes.string,

  /** xPosition for Dropdown menu */
  xPosition: PropTypes.string,

  /** Items list to display in the popup menu */
  items: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string,
      title: PropTypes.string,
      colors: PropTypes.shape({
        title: PropTypes.string,
        iconHover: PropTypes.string,
      }),
    }),
  ).isRequired,

  /** onOpen function to fire when the Dropdown menu is open */
  onOpen: PropTypes.func,
}

// @ts-expect-error TS(2339) FIXME: Property 'defaultProps' does not exist on type 'ty... Remove this comment to see the full error message
DropdownMenu.defaultProps = {
  ariaLabelPopup: null,
  xPosition: 'right',
  onOpen: null,
}
