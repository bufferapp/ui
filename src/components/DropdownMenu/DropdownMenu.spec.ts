// @ts-expect-error TS(7016) FIXME: Could not find a declaration file for module 'jest... Remove this comment to see the full error message
import snap from 'jest-auto-snapshots'
import DropdownMenu from './DropdownMenu'

snap(DropdownMenu, './DropdownMenu.tsx')
