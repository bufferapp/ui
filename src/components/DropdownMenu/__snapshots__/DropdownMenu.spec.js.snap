// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest-auto-snapshots > DropdownMenu Matches snapshot when array prop "items" has one item: "[object Object]" 1`] = `
<ForwardRef(styled.ul)
  aria-label="jest-auto-snapshots String Fixture"
  role="menubar"
  usingMouse={false}
>
  <ForwardRef(styled.li)
    menuOption={true}
    role="none"
  >
    <NodeFixture
      aria-expanded={false}
      aria-haspopup="true"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="menuitem"
      tabIndex="0"
    />
    <PopupMenu
      aria-label="jest-auto-snapshots String Fixture"
      ariaLabelPopup={null}
      closePopup={[Function]}
      horizontalOffset={null}
      isOpen={false}
      items={
        Array [
          Object {
            "colors": Object {
              "iconHover": "jest-auto-snapshots String Fixture",
              "title": "jest-auto-snapshots String Fixture",
            },
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          },
        ]
      }
      onBlur={[Function]}
      onOpen={[MockFunction]}
      role="menu"
      usingMouse={false}
      xPosition="jest-auto-snapshots String Fixture"
    />
  </ForwardRef(styled.li)>
</ForwardRef(styled.ul)>
`;

exports[`jest-auto-snapshots > DropdownMenu Matches snapshot when passed all props 1`] = `
<ForwardRef(styled.ul)
  aria-label="jest-auto-snapshots String Fixture"
  role="menubar"
  usingMouse={false}
>
  <ForwardRef(styled.li)
    menuOption={true}
    role="none"
  >
    <NodeFixture
      aria-expanded={false}
      aria-haspopup="true"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="menuitem"
      tabIndex="0"
    />
    <PopupMenu
      aria-label="jest-auto-snapshots String Fixture"
      ariaLabelPopup={null}
      closePopup={[Function]}
      horizontalOffset={null}
      isOpen={false}
      items={
        Array [
          Object {
            "colors": Object {
              "iconHover": "jest-auto-snapshots String Fixture",
              "title": "jest-auto-snapshots String Fixture",
            },
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          },
        ]
      }
      onBlur={[Function]}
      onOpen={[MockFunction]}
      role="menu"
      usingMouse={false}
      xPosition="jest-auto-snapshots String Fixture"
    />
  </ForwardRef(styled.li)>
</ForwardRef(styled.ul)>
`;

exports[`jest-auto-snapshots > DropdownMenu Matches snapshot when passed only required props 1`] = `
<ForwardRef(styled.ul)
  aria-label="jest-auto-snapshots String Fixture"
  role="menubar"
  usingMouse={false}
>
  <ForwardRef(styled.li)
    menuOption={true}
    role="none"
  >
    <NodeFixture
      aria-expanded={false}
      aria-haspopup="true"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="menuitem"
      tabIndex="0"
    />
    <PopupMenu
      aria-label={null}
      ariaLabelPopup={null}
      closePopup={[Function]}
      horizontalOffset={null}
      isOpen={false}
      items={
        Array [
          Object {
            "colors": Object {
              "iconHover": "jest-auto-snapshots String Fixture",
              "title": "jest-auto-snapshots String Fixture",
            },
            "id": "jest-auto-snapshots String Fixture",
            "title": "jest-auto-snapshots String Fixture",
          },
        ]
      }
      onBlur={[Function]}
      onOpen={null}
      role="menu"
      usingMouse={false}
      xPosition="right"
    />
  </ForwardRef(styled.li)>
</ForwardRef(styled.ul)>
`;
